src/components/TextareaAutosize.vue

```diff
@@ -1,121 +1,122 @@
-          }
-        }
-        const heightVal = contentHeight + 'px';
-        height.value = `${heightVal}${important ? ' !important' : ''}`;
-      });
-      return this;
-    }
+  autosize: {
+    type: Boolean,
+    default: true
+  },
+  minHeight: {
+    type: [Number],
+    default: null
+  },
+  maxHeight: {
+    type: [Number],
+    default: null
+  },
+  important: {
+    type: [Boolean, Array],
+    default: false
+  }
+});
 
-    watch(modelValue, v => {
-      val.value = v;
-    });
+const emit = defineEmits(['update:modelValue']);
 
-    watch(val, v => {
-      nextTick(resize);
-      emit('update:modelValue', v);
-    });
+const { modelValue, minHeight, maxHeight } = toRefs(props);
 
-    watch(minHeight, () => {
-      nextTick(resize);
-    });
+// data property for v-model binding with real textarea tag
+const val = ref(props.modelValue);
+// works when content height becomes more then value of the maxHeight property
+const maxHeightScroll = ref(false);
+const height = ref('auto');
+const textarea = ref(null);
 
-    watch(maxHeight, () => {
-      nextTick(resize);
-    });
+const isResizeImportant = computed(() => {
+  const imp = props.important;
+  return imp === true || (Array.isArray(imp) && imp.includes('resize'));
+});
 
-    watch(val, v => {
-      if (v) resize();
-    });
+const isOverflowImportant = computed(() => {
+  const imp = props.important;
+  return imp === true || (Array.isArray(imp) && imp.includes('overflow'));
+});
 
-    onMounted(() => resize());
+const isHeightImportant = computed(() => {
+  const imp = props.important;
+  return imp === true || (Array.isArray(imp) && imp.includes('height'));
+});
 
-    return { val, computedStyles, textarea, resize };
-  }
-};
+const computedStyles = computed(() => {
+  if (!props.autosize) return {};
+  return {
+    resize: !isResizeImportant.value ? 'none' : 'none !important',
+    height: height.value,
+    overflow: maxHeightScroll.value
+      ? 'auto'
+      : !isOverflowImportant.value
+      ? 'hidden'
+      : 'hidden !important'
+  };
+});
+
+function resize() {
+  const important = isHeightImportant.value ? 'important' : '';
+  height.value = `auto${important ? ' !important' : ''}`;
+  nextTick(() => {
+    let contentHeight = textarea.value.scrollHeight + 1;
+    if (props.minHeight) {
+      contentHeight =
+        contentHeight < props.minHeight ? props.minHeight : contentHeight;
+    }
+    if (props.maxHeight) {
+      if (contentHeight > props.maxHeight) {
+        contentHeight = props.maxHeight;
+        maxHeightScroll.value = true;
+      } else {
+        maxHeightScroll.value = false;
+      }
+    }
+    const heightVal = contentHeight + 'px';
+    height.value = `${heightVal}${important ? ' !important' : ''}`;
+  });
+  return this;
+}
+
+watch(modelValue, v => {
+  val.value = v;
+});
+
+watch(val, v => {
+  nextTick(resize);
+  emit('update:modelValue', v);
+});
+
+watch(minHeight, () => {
```

Comment:
You could watch multiple source here like this 
```suggestion
watch([minHeight, maxHeight], () => {
```
See https://v3.vuejs.org/guide/reactivity-computed-watchers.html#watching-multiple-sources